source/draw/gpu/opengl/OpenGLCanvas.ooc

```diff
@@ -32,16 +32,10 @@ OpenGLCanvas: class extends OpenGLSurface {
 		gpuMap: GpuMap = drawState map as GpuMap ?? this context defaultMap
 		viewport := (drawState viewport hasZeroArea) ? IntBox2D new(this size) : drawState viewport
 		this context backend setViewport(viewport)
-		gpuMap view = _toLocal * drawState getTransformNormalized() normalizedToReference(this size) * _toLocal
-		if (this _focalLength > 0.0f) {
-			a := 2.0f * this _focalLength / this size x
-			f := -(this _coordinateTransform e as Float) * 2.0f * this _focalLength / this size y
-			k := (this _farPlane + this _nearPlane) / (this _farPlane - this _nearPlane)
-			o := 2.0f * this _farPlane * this _nearPlane / (this _farPlane - this _nearPlane)
-			gpuMap projection = FloatTransform3D new(a, 0.0f, 0.0f, 0.0f, 0.0f, f, 0.0f, 0.0f, 0.0f, 0.0f, k, -1.0f, 0.0f, 0.0f, o, 0.0f)
-		} else
-			gpuMap projection = FloatTransform3D createScaling(2.0f / this size x, -(this _coordinateTransform e as Float) * 2.0f / this size y, 1.0f)
-		gpuMap model = this _createModelTransform(IntBox2D new(this size))
+		gpuMap model = FloatTransform3D identity // TODO: set using a destination argument in drawState
+		gpuMap view = _toLocal * drawState getTransformNormalized() * _toLocal
```

Comment:
The transform is flipped by applying the transform between two flip matrices that are their own inverses.
We multiply in reverse order when using inverse matrices because the coordinate system of each side is flipped but the order does not matter here as both expressions are equivalent.